### PR TITLE
chore: bump gunicorn to 22.0.0

### PR DESCRIPTION
**Pull request by dpgaspar**
### SUMMARY
Bump gunicorn to address a potencial vulnerability

https://docs.gunicorn.org/en/latest/news.html#id1

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by mistercrunch — APPROVED** [review]
LGTM after reviewing breaking changes https://docs.gunicorn.org/en/latest/news.html

Looks ok to me but thought I'd paste here for @dpgaspar to have another look pre-merge:

** Breaking changes **

minimum version is Python 3.7

the limitations on valid characters in the HTTP method have been bounded to Internet Standards

requests specifying unsupported transfer coding (order) are refused by default (rare)

HTTP methods are no longer casefolded by default (IANA method registry contains none affected)

HTTP methods containing the number sign (#) are no longer accepted by default (rare)

HTTP versions < 1.0 or >= 2.0 are no longer accepted by default (rare, only HTTP/1.1 is supported)

HTTP versions consisting of multiple digits or containing a prefix/suffix are no longer accepted

HTTP header field names Gunicorn cannot safely map to variables are silently dropped, as in other software

HTTP headers with empty field name are refused by default (no legitimate use cases, used in exploits)

requests with both Transfer-Encoding and Content-Length are refused by default (such a message might indicate an attempt to perform request smuggling)

empty transfer codings are no longer permitted (reportedly seen with really old & broken proxies)

File: pyproject.toml [hunk 1]
```diff
@@ -55,7 +55,7 @@ dependencies = [
     "flask-wtf>=1.1.0, <2.0",
     "func_timeout",
     "geopy",
-    "gunicorn>=21.2.0, <22.0; sys_platform != 'win32'",
+    "gunicorn>=22.0.0; sys_platform != 'win32'",
```
Comment (john-bodley) [inline pyproject.toml:58]: @dpgaspar should we include an upper limit on `<23.0`?

Comment (dpgaspar) [reply pyproject.toml:58]: We may be moving away from always pinning the upper limit: https://github.com/apache/superset/pull/28144#discussion_r1572740500

**Comment by dpgaspar** [comment]
> LGTM after reviewing breaking changes https://docs.gunicorn.org/en/latest/news.html
> 
> Looks ok to me but thought I'd paste here for @dpgaspar to have another look pre-merge:
> 
> ** Breaking changes **
> 
> minimum version is Python 3.7
> 
> the limitations on valid characters in the HTTP method have been bounded to Internet Standards
> 
> requests specifying unsupported transfer coding (order) are refused by default (rare)
> 
> HTTP methods are no longer casefolded by default (IANA method registry contains none affected)
> 
> HTTP methods containing the number sign (#) are no longer accepted by default (rare)
> 
> HTTP versions < 1.0 or >= 2.0 are no longer accepted by default (rare, only HTTP/1.1 is supported)
> 
> HTTP versions consisting of multiple digits or containing a prefix/suffix are no longer accepted
> 
> HTTP header field names Gunicorn cannot safely map to variables are silently dropped, as in other software
> 
> HTTP headers with empty field name are refused by default (no legitimate use cases, used in exploits)
> 
> requests with both Transfer-Encoding and Content-Length are refused by default (such a message might indicate an attempt to perform request smuggling)
> 
> empty transfer codings are no longer permitted (reportedly seen with really old & broken proxies)

I took a look, seems unlikely to break anyones deployments

**Comment by denodo-research-labs** [comment]
[CVE-2024-1135](https://nvd.nist.gov/vuln/detail/CVE-2024-1135) is reported as `HIGH` by some security tools like PRISMA.

Will this version bump be applied to the 3.1-x branch too to avoid that CVE affecting current stable versions?

**Comment by g-jgreen** [comment]
Any update on this? Seems to not be available on release 4.0.2. 

Thanks

**Comment by mistercrunch** [comment]
Should be in 4.1.0, which has an open release candidate

**Pull request merged**